cpp/Platform.Disposables/DisposableBase.h

```diff
@@ -26,29 +41,34 @@
             return false;
         }
 
-        static DisposableBase() { std::atexit(OnProcessExit); }
 
-        protected: DisposableBase()
+        public: DisposableBase()
         {
+            std::shared_ptr<DisposableBase> this_ptr = std::shared_ptr<DisposableBase>(this);
+            std::weak_ptr<DisposableBase> this_weak = std::weak_ptr<DisposableBase>(this_ptr);
+
+
             _disposed = 0;
-            _disposablesWeekReferencesStack.Push(WeakReference<DisposableBase>(this, false));
+            _disposablesWeekReferencesStack.push(this_weak);
+            std::atexit(OnProcessExit);
         }
 
-        ~DisposableBase() { Destruct(); }
+        //TODO: завязывайте с вызовом всякой виртуальщины в конструкторе/деструкторе
+        //~DisposableBase() { Destruct(); }
```

Comment:
Because clang says "Inspection' Virtual call from constructor or destructor' "

In addition, you can look at these two codes:
https://rextester.com/PUOW20883 (С++)
https://rextester.com/TWL32777 (C#)

In principle, this problem can be easily solved, but I'm not sure that it will be in the style of C# Disposables
For example, something like this:
```cpp
class DisposableWrapper {
    DisposableBase* disposableObject;
    
public: 
    
    DisposableWrapper(DisposableBase* disposableObject) : disposableObject(disposableObject) {}
    
    ~DisposableWrapper() {
        if(disposableObject != null) 
            disposableObject->Dispose(...);
        else 
            .....
            
        delete disposableObject;
    }
};
```

Reply:
Ok, we can skip it at the moment, and force child implementations to call `Destruct` method in the destructor.

Reply:
I'll try to figure it out